pkg/yurtctl/cmd/markautonomous/markautonomous.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2020 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package markautonomous
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/alibaba/openyurt/pkg/yurtctl/constants"
+	kubeutil "github.com/alibaba/openyurt/pkg/yurtctl/util/kubernetes"
+)
+
+// MarkAutonomousOptions has the information that required by convert operation
+type MarkAutonomousOptions struct {
+	*kubernetes.Clientset
+	AutonomousNodes  []string
+	MarkAllEdgeNodes bool
+}
+
+// NewMarkAutonomousOptions creates a new MarkAutonomousOptions
+func NewMarkAutonomousOptions() *MarkAutonomousOptions {
+	return &MarkAutonomousOptions{}
+}
+
+// NewMarkAutonomousCmd generates a new markautonomous command
+func NewMarkAutonomousCmd() *cobra.Command {
+	co := NewMarkAutonomousOptions()
+	cmd := &cobra.Command{
+		Use:   "markautonomous -a AUTONOMOUSNODES",
+		Short: "mark the nodes as autonomous",
+		Run: func(cmd *cobra.Command, _ []string) {
+			if err := co.Complete(cmd.Flags()); err != nil {
+				klog.Fatalf("fail to complete the markautonomous option: %s", err)
+			}
+			if err := co.Validate(); err != nil {
+				klog.Fatalf("markautonomous option is invalid: %s", err)
+			}
+			if err := co.RunMarkAutonomous(); err != nil {
+				klog.Fatalf("fail to make nodes autonomous: %s", err)
+			}
+		},
+	}
+
+	cmd.Flags().Bool("mark-all-edge-nodes", true,
```

Comment:
Can we remove this flag? If autonomous-nodes string is empty, we assume it means all edge nodes.

Reply:
removed